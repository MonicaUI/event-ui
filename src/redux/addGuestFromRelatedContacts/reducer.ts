const TOGGLE_RELATED_CONTACTS_VIEW = 'event-guestside-site/addGuestFromRelatedContacts/LIST_RELATED_CONTACTS';
const LOAD_RELATED_CONTACTS_FOR_INVITEE =
  'event-guestside-site/addGuestFromRelatedContacts/LOAD_RELATED_CONTACTS_FOR_INVITEE';
const LOAD_RELATED_CONTACTS_SEARCH_RESULTS =
  'event-guestside-site/addGuestFromRelatedContacts/LOAD_RELATED_CONTACTS_SEARCH_RESULTS';

export function setRelatedContacts(contactId: $TSFixMe, relatedContacts: $TSFixMe): $TSFixMe {
  return { type: LOAD_RELATED_CONTACTS_FOR_INVITEE, payload: { contactId, relatedContacts } };
}

export function toggleRelatedContacts(): $TSFixMe {
  return { type: TOGGLE_RELATED_CONTACTS_VIEW };
}

export function setRelatedContactsSearch(
  contactId: $TSFixMe,
  searchCriteria: $TSFixMe,
  searchResults: $TSFixMe
): $TSFixMe {
  return { type: LOAD_RELATED_CONTACTS_SEARCH_RESULTS, payload: { contactId, searchCriteria, searchResults } };
}

export default function addGuestFromRelatedContactsReducer(state = {}, action: $TSFixMe): $TSFixMe {
  switch (action.type) {
    case LOAD_RELATED_CONTACTS_FOR_INVITEE: {
      return {
        ...state,
        [action.payload.contactId]: {
          ...state[action.payload.contactId],
          relatedContacts: action.payload.relatedContacts
        }
      };
    }
    case TOGGLE_RELATED_CONTACTS_VIEW: {
      return {
        ...state,
        showRelatedContactsView: !(state as $TSFixMe).showRelatedContactsView
      };
    }
    case LOAD_RELATED_CONTACTS_SEARCH_RESULTS: {
      return {
        ...state,
        [action.payload.contactId]: {
          ...state[action.payload.contactId],
          relatedContactsSearchCriteria: action.payload.searchCriteria,
          relatedContactsSearchResults: action.payload.searchResults
        }
      };
    }
    default: {
      return state;
    }
  }
}
