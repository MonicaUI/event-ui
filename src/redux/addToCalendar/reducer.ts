import { CALENDAR_EVENT_ID } from '../../utils/addToCalendarUtils';

export const LOAD_ADD_TO_CALENDAR_PROVIDERS = 'event-guestside-site/calendar/LOAD_CALENDAR_PROVIDERS';
export const LOAD_EVENT_CALENDAR = 'event-guestside-site/calendar/LOAD_EVENT_CALENDAR';
export const LOAD_SESSION_CALENDAR = 'event-guestside-site/calendar/LOAD_SESSION_CALENDAR';

export function setAddToCalendarProviders(calendarProviders: $TSFixMe): $TSFixMe {
  return {
    type: LOAD_ADD_TO_CALENDAR_PROVIDERS,
    payload: { calendarProviders }
  };
}

export function setCalendarUrl(calendarType: $TSFixMe, calendarUrl: $TSFixMe, entityId = CALENDAR_EVENT_ID): $TSFixMe {
  if (entityId === CALENDAR_EVENT_ID) {
    return {
      type: LOAD_EVENT_CALENDAR,
      payload: { calendarType, calendarUrl }
    };
  }

  return {
    type: LOAD_SESSION_CALENDAR,
    payload: { calendarType, calendarUrl, entityId }
  };
}

export default function addToCalendarReducer(state = {}, action: $TSFixMe): $TSFixMe {
  switch (action.type) {
    case LOAD_ADD_TO_CALENDAR_PROVIDERS: {
      return {
        ...state,
        ...action.payload.calendarProviders
      };
    }
    case LOAD_EVENT_CALENDAR: {
      return {
        ...state,
        [action.payload.calendarType]: {
          ...state[action.payload.calendarType],
          [CALENDAR_EVENT_ID]: action.payload.calendarUrl
        }
      };
    }
    case LOAD_SESSION_CALENDAR: {
      return {
        ...state,
        [action.payload.calendarType]: {
          ...state[action.payload.calendarType],
          [action.payload.entityId]: action.payload.calendarUrl
        }
      };
    }
    default: {
      return state;
    }
  }
}
