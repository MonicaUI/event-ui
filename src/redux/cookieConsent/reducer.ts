// Reducer to update state for cookie Consent
import { DENY } from '../../utils/CookieConstants';

export const SET_COOKIE_CONSENT = 'event-guestside-site/cookieConsent/SET_COOKIE_CONSENT';

export default function cookieConsentReducer(state = { status: DENY }, action: $TSFixMe): $TSFixMe {
  switch (action.type) {
    case SET_COOKIE_CONSENT:
      return {
        ...state,
        status: action.payload.cookieStatus
      };
    default:
      return state;
  }
}
