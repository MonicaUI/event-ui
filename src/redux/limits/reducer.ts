const SET_ACCOUNT_LIMITS = 'event-guestside-site/limits/SET_ACCOUNT_LIMITS';

export const setAccountLimits = (accountLimits: $TSFixMe): $TSFixMe => {
  return {
    type: SET_ACCOUNT_LIMITS,
    payload: {
      accountLimits
    }
  };
};

export default function reducer(state: $TSFixMe, action: $TSFixMe): $TSFixMe {
  switch (action.type) {
    case SET_ACCOUNT_LIMITS: {
      return action.payload.accountLimits;
    }
    default: {
      return state;
    }
  }
}
