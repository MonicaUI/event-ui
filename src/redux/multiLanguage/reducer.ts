import { SET_LOCALE } from 'nucleus-widgets/redux/modules/text';
import { LOAD_LANGUAGE } from './actions';

export default function reducer(state: $TSFixMe, action: $TSFixMe): $TSFixMe {
  switch (action.type) {
    case SET_LOCALE:
      return {
        ...state,
        locale: action.payload.locale
      };
    case LOAD_LANGUAGE:
      return {
        ...state,
        loadedLanguages: [...state.loadedLanguages, action.payload]
      };
    default:
      return state;
  }
}
