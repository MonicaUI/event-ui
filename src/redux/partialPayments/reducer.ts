import { UPDATE_PAYMENT_AMOUNT, UPDATE_SELECTED_PAYMENT_OPTION } from './actions';
import { PAYMENT_AMOUNT_OPTION } from 'event-widgets/utils/paymentConstant';
import { START_MODIFICATION_SUCCESS } from '../registrationForm/regCart/actionTypes';
import { GET_ORDERS } from '../actionTypes';

// Reducer to update state for Partial Payments
export default function reducer(state: $TSFixMe, action: $TSFixMe): $TSFixMe {
  const defaultPartialPayment = { paymentAmountOption: null, paymentAmount: null };
  switch (action.type) {
    case UPDATE_SELECTED_PAYMENT_OPTION: {
      const FULL_PAYMENT_VALUE = PAYMENT_AMOUNT_OPTION.FULL_PAYMENT.value;
      const paymentOption =
        action.payload === FULL_PAYMENT_VALUE
          ? PAYMENT_AMOUNT_OPTION.FULL_PAYMENT
          : PAYMENT_AMOUNT_OPTION.PARTIAL_PAYMENT;
      return {
        ...state,
        paymentAmountOption: paymentOption
      };
    }
    case UPDATE_PAYMENT_AMOUNT:
      return {
        ...state,
        paymentAmount: action.payload
      };
    case START_MODIFICATION_SUCCESS:
    case GET_ORDERS:
      return defaultPartialPayment;
    default:
      return state;
  }
}
