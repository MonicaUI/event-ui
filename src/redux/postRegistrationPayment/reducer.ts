import {
  RESET_SUBMIT_WEB_PAYMENT,
  SUBMIT_WEB_PAYMENT,
  STORE_WEBPAYMENT_DATA,
  FINALIZE_POSTREG_CHECKOUT_PENDING,
  FINALIZE_POSTREG_CHECKOUT_COMPLETE
} from './actionTypes';
import { PAYMENT_TYPE } from 'event-widgets/constants/PaymentType';
import { addQueryParams } from '../../utils/queryUtils';
import { PAYMENT_AMOUNT_OPTION } from 'event-widgets/utils/paymentConstant';

export default function reducer(state = '', action: $TSFixMe): $TSFixMe {
  switch (action.type) {
    case SUBMIT_WEB_PAYMENT: {
      return {
        // @ts-expect-error ts-migrate(2698) FIXME: Spread types may only be created from object types... Remove this comment to see the full error message
        ...state,
        submitWebPayment: true
      };
    }
    case RESET_SUBMIT_WEB_PAYMENT:
      return {
        // @ts-expect-error ts-migrate(2698) FIXME: Spread types may only be created from object types... Remove this comment to see the full error message
        ...state,
        submitWebPayment: false
      };
    case STORE_WEBPAYMENT_DATA:
      return {
        // @ts-expect-error ts-migrate(2698) FIXME: Spread types may only be created from object types... Remove this comment to see the full error message
        ...state,
        webPaymentData: action.payload.webPaymentData
      };
    case FINALIZE_POSTREG_CHECKOUT_PENDING:
      return {
        // @ts-expect-error ts-migrate(2698) FIXME: Spread types may only be created from object types... Remove this comment to see the full error message
        ...state,
        isCheckingOut: true
      };
    case FINALIZE_POSTREG_CHECKOUT_COMPLETE:
      return {
        // @ts-expect-error ts-migrate(2698) FIXME: Spread types may only be created from object types... Remove this comment to see the full error message
        ...state,
        isCheckingOut: false
      };
    default:
      return {
        // @ts-expect-error ts-migrate(2698) FIXME: Spread types may only be created from object types... Remove this comment to see the full error message
        ...state,
        submitWebPayment: false
      };
  }
}

function getPaymentDistributions(orders) {
  const paymentDistribution = new Map();
  orders.forEach(order => {
    if (order.orderType === 'OfflineCharge') {
      order.orderItems.forEach(item => {
        if (item.amountDue > 0) {
          if (paymentDistribution.get(order.attendeeId)) {
            paymentDistribution.get(order.attendeeId).orders.push({
              orderId: order.orderId,
              orderDetailId: item.itemId,
              amountIntended: item.amountDue
            });
          } else {
            paymentDistribution.set(order.attendeeId, {
              attendeeId: order.attendeeId,
              orders: [
                {
                  orderId: order.orderId,
                  orderDetailId: item.itemId,
                  amountIntended: item.amountDue
                }
              ]
            });
          }
        }
      });
    }
  });
  return Array.from(paymentDistribution.values());
}

export function getTotals(orders: $TSFixMe): $TSFixMe {
  let subtotal = 0;
  let total = 0;
  if (orders) {
    orders.forEach(order => {
      if (order.orderType === 'OfflineCharge') {
        order.orderItems.forEach(item => {
          total = +(total + item.amountDue).toFixed(12);
          if (
            item.productType !== 'Tax' &&
            item.productType !== 'RegistrationTypeServiceFee' &&
            item.productType !== 'PaymentTypeServiceFee'
          ) {
            subtotal = +(subtotal + item.amountDue).toFixed(12);
          }
        });
      }
    });
  }
  return { subtotal, total };
}

export function getPostRegPaymentPricingInfo(
  orders: $TSFixMe,
  webPaymentData: $TSFixMe,
  accessToken: $TSFixMe,
  partialPaymentSetting?: $TSFixMe
): $TSFixMe {
  const token = accessToken.split(' ');
  const totalAmount =
    partialPaymentSetting?.paymentAmountOption === PAYMENT_AMOUNT_OPTION.PARTIAL_PAYMENT
      ? partialPaymentSetting.paymentAmount
      : getTotals(orders).total;
  return {
    postRegPaymentData: {
      totalAmount,
      cardType: webPaymentData.cardType,
      paymentDistribution: getPaymentDistributions(orders)
    },
    continueCheckout: true,
    checkoutInfo: {
      paymentMethodType: webPaymentData.cardType,
      paymentType: PAYMENT_TYPE.ONLINE,
      paymentAmount: totalAmount,
      paymentDetail: {
        amount: totalAmount,
        paymentMethodMode: 'UseNewPaymentMethod',
        browserLandingURL: addQueryParams(window.location.href, { fromRedirect: 'true' }).replace(/(^\w+:|^)\/\//, ''),
        contextId: webPaymentData.contextId,
        coreBearerToken: token[1]
      }
    }
  };
}

/**
 * This method is called only for Service Fees with Post Reg Payments
 * to pass checkout info object when startRegCartCheckout is called for
 * continuing checkout
 */
export function getCheckoutInfoForPostServiceFees(
  webPaymentData: $TSFixMe,
  regCartPricing: $TSFixMe,
  accessToken: $TSFixMe
): $TSFixMe {
  const token = accessToken.split(' ');
  const totalAmount = regCartPricing.netFeeAmountChargeWithPaymentAmountServiceFee;
  return {
    paymentMethodType: webPaymentData.cardType,
    paymentType: PAYMENT_TYPE.ONLINE,
    paymentAmount: totalAmount,
    paymentDetail: {
      amount: totalAmount,
      paymentMethodMode: 'UseNewPaymentMethod',
      browserLandingURL: addQueryParams(window.location.href, { fromRedirect: 'true' }).replace(/(^\w+:|^)\/\//, ''),
      contextId: webPaymentData.contextId,
      coreBearerToken: token[1]
    }
  };
}
