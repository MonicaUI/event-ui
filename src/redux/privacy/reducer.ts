import { SWITCH_CCPA_DIALOG_MODE } from '../actionTypes';
import { DIALOG_MODE } from 'event-widgets/utils/ccpaUtils';

export default function reducer(
  state = {
    ccpa: { dialogMode: DIALOG_MODE.UNREGISTERED }
  },
  action: $TSFixMe
): $TSFixMe {
  switch (action.type) {
    case SWITCH_CCPA_DIALOG_MODE:
      return {
        ...state,
        ccpa: {
          ...state.ccpa,
          dialogMode: action.payload.dialogMode
        }
      };
    default:
      return state;
  }
}
