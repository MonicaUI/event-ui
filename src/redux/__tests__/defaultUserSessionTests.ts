import reducer, { getUserType } from '../defaultUserSession';

test('Verify initial state', () => {
  expect(reducer(undefined, {})).toMatchSnapshot();
});

test('getUserType', () => {
  expect(getUserType({ isPlanner: true })).toBe('planner');
  expect(getUserType({ isPreview: true })).toBe('preview');
  expect(getUserType({})).toBe('standard');
  expect(getUserType()).toBe('standard');
});
