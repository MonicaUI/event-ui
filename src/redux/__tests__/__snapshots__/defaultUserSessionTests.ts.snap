// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Verify initial state 1`] = `
Object {
  "defaultRegPackId": "",
  "eventBuildWizardExitUrl": "",
  "eventId": "",
  "freeTrialPurchaseCta": "",
  "httpReferrer": "",
  "isFreeTrial": false,
  "isPlanner": false,
  "isPreview": false,
  "isTestMode": false,
  "licenseTypeId": 1,
  "showEventBuildWizardBanner": false,
}
`;
